src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -268,9 +260,13 @@ private void waitForState(State target) {
     }
 
     private void send(ControlMessage message) {
-        if (getState() != State.Connected) {
+        if (getState() == State.Disconnected) {
```

Comment:
We should not attempt to connect if an existing connection attempt (`Sleeping` or `Connecting`) is underway.